Sources/SwiftCoAP/SCServer.swift

```diff
@@ -165,7 +165,9 @@ class SCServer: NSObject {
                 setupReliableTransmissionOfMessage(notification, forResource: resource)
             }
         }
-        delegate?.swiftCoapServer(self, willUpdatedObserversForResource: resource)
+        DispatchQueue.main.async {
```

Comment:
Why the main thread here? I'd expect this layer to be on some background thread and the consumer doing additional work before surfacing to the UI (and the main thread).

But OK for now, we can figure it out later.